common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+
+    int group_idx = -1;
+    for (int i=0; i<ARRAY_SIZE(GPIO_GROUP_NAME_LST); i++) {
+        if (!strcmp(dev_name, GPIO_GROUP_NAME_LST[i]))
+            group_idx = i;
+    }
+
+    return group_idx;
+}
+
+static const char* gpio_get_name(const char *dev_name, int pin_num)
+{
+    if (!dev_name)
+        return NULL;
+
+    int name_idx = -1;
+    name_idx = pin_num + 32 * gpio_get_group_idx_by_dev_name(dev_name);
+
+    if (name_idx == -1)
+        return NULL;
+
+    if (name_idx >= GPIO_CFG_SIZE)
+        return "Undefined";
+
+    return gpio_name[name_idx];
+}
+#endif
+
+/* 
+    Command SENSOR 
+*/
+#if COMMAND_SENSOR == 1
+static bool access_check(uint8_t sensor_num) {
+    bool (*access_checker)(uint8_t);
+    access_checker = sensor_config[SnrNum_SnrCfg_map[sensor_num]].access_checker;
+
+    return (access_checker)(sensor_config[SnrNum_SnrCfg_map[sensor_num]].num);
+}
+
+static int sensor_get_idx_by_snr_num(uint16_t sensor_num) {
+    for (int sen_idx=0; sen_idx<SDR_NUM; sen_idx++) {
+        if (sensor_num == sensor_config[sen_idx].num)
+            return sen_idx;
+    }
+
+    return -1;
+}
+
+static int sensor_access(const struct shell *shell, int snr_num, enum SENSOR_ACCESS mode) {
+    if (!shell)
+        return 1;
+
+    if (snr_num >= SENSOR_NUM_MAX || snr_num < 0)
+        return 1;
+
+    switch (mode) {
+        /* Get sensor info by "sensor_config" table */
+        case SENSOR_READ:
+            ;
+            int sen_idx = sensor_get_idx_by_snr_num(snr_num);
+            if (sen_idx == -1) {
+                shell_error(shell, "No such sensor number!");
+                return 1;
+            }
+            char *check_access = (access_check(sensor_config[sen_idx].num) == true) ? "O":"X";
+            shell_print(shell, "[0x%-2x] %-35s: %-5s | access[%s] | %-20s | %-8d",
+                sensor_config[sen_idx].num,
+                "Unsupport name",
+                sensor_type_name[sensor_config[sen_idx].type],
+                check_access,
+                sensor_status_name[sensor_config[sen_idx].cache_status],
+                sensor_config[sen_idx].cache);
+            break;
+
+        case SENSOR_WRITE:
+            /* TODO */
+            break;
+
+        default:
+            break;
+    }
+
+    return 0;
+}
+#endif
+
+/*********************************************************************************************************
+ * COMMAND FUNCTION SECTION
+**********************************************************************************************************/
+/* 
+    Command header 
+*/
+static int cmd_info_print(const struct shell *shell, size_t argc, char **argv)
+{
+    shell_print(shell, "========================{SHELL COMMAND INFO}========================================");
+    shell_print(shell, "* NAME:          Platform command");
+    shell_print(shell, "* DESCRIPTION:   Commands that could be used to debug or validate.");
+    shell_print(shell, "* AUTHOR:        MouchenHung");
```

Comment:
Probably leave out author as this code might be reused or changed by others and moved elsewhere.
git log/blame will show original author and the info isn't needed for users.

Reply:
Got it, this suggestion would add in next pr, thanks!